libbeat/processors/condition.go

```diff
@@ -269,26 +269,36 @@ func (c *Condition) checkEquals(event common.MapStr) bool {
 }
 
 func (c *Condition) checkContains(event common.MapStr) bool {
-
+outer:
 	for field, equalValue := range c.contains {
-
 		value, err := event.GetValue(field)
 		if err != nil {
 			return false
 		}
 
-		sValue, err := extractString(value)
-		if err != nil {
-			logp.Warn("unexpected type %T in contains condition as it accepts only strings. ", value)
+		switch value.(type) {
+		case string:
+			if !strings.Contains(value.(string), equalValue) {
+				return false
+			}
+		case *string:
+			if !strings.Contains(*value.(*string), equalValue) {
+				return false
+			}
+		case []string:
+			for _, s := range value.([]string) {
+				if strings.Contains(s, equalValue) {
+					continue outer
```

Comment:
That would be changing the behavior of the method. This method returns true iff all the `contains` conditions are satisfied. Returning true would cause the method to not evaluate any remaining conditions.

When I was adding this, I initially did `return true` only to realize I broken a test.